tracing-test-utils/src/main/java/com/palantir/tracing/SpanRenderer.java

```diff
@@ -0,0 +1,382 @@
+                    renderDuration(transposedStartMicros + TimeUnit.MICROSECONDS.convert(
+                            span.getDurationNanoSeconds(),
+                            TimeUnit.NANOSECONDS), TimeUnit.MICROSECONDS),
+                    span.getOperation(),
+                    renderDuration(span.getDurationNanoSeconds(), TimeUnit.NANOSECONDS),
+                    suspectedCollision ? " (collision)" : "");
+        }
+
+
+        public void rawSpanJson(Collection<Span> spans, StringBuilder sb) {
+            sb.append("\n<pre style=\"background: #CED9E0;"
+                    + "color: #738694;"
+                    + "padding: 30px;"
+                    + "overflow-x: scroll;"
+                    + "margin-top: 100px;\">");
+            spans.stream().sorted(Comparator.comparingLong(Span::getStartTimeMicroSeconds)).forEach(s -> {
+                try {
+                    sb.append('\n');
+                    sb.append(writer.writeValueAsString(s));
+                } catch (JsonProcessingException e) {
+                    throw new RuntimeException("Unable to JSON serialize span " + s, e);
+                }
+            });
+            sb.append("\n</pre>");
+        }
+    }
+
+    private static class AsciiFormatter {
+        private final TimeBounds bounds;
+
+        AsciiFormatter(TimeBounds bounds) {
+            this.bounds = bounds;
+        }
+
+        public String formatSpan(Span span) {
+            long transposedStartMicros = span.getStartTimeMicroSeconds() - bounds.startMicros();
+            float leftPercentage = percentage(transposedStartMicros, bounds.durationMicros());
+            float widthPercentage = percentage(span.getDurationNanoSeconds(), bounds.durationNanos());
+
+            int numSpaces = (int) Math.floor(leftPercentage);
+            int numHashes = (int) Math.floor(widthPercentage);
+
+            String spaces = Strings.repeat(" ", numSpaces);
+
+            String name = span.getOperation().substring(0, Math.min(numHashes, span.getOperation().length()));
+            String hashes = name + Strings.repeat("-", numHashes - name.length());
+
+            return spaces + (hashes.isEmpty() ? "|" : hashes);
+        }
+    }
+
+    private static String renderDuration(float amount, TimeUnit timeUnit) {
+        ImmutableMap<TimeUnit, TimeUnit> largerUnit = ImmutableMap.<TimeUnit, TimeUnit>builder()
+                .put(TimeUnit.NANOSECONDS, TimeUnit.MICROSECONDS)
+                .put(TimeUnit.MICROSECONDS, TimeUnit.MILLISECONDS)
+                .put(TimeUnit.MILLISECONDS, TimeUnit.SECONDS)
+                .build();
+
+        ImmutableMap<TimeUnit, String> abbreviation = ImmutableMap.<TimeUnit, String>builder()
+                .put(TimeUnit.NANOSECONDS, "ns")
+                .put(TimeUnit.MICROSECONDS, "micros")
+                .put(TimeUnit.MILLISECONDS, "ms")
+                .put(TimeUnit.SECONDS, "s")
+                .build();
```

Comment:
nit: These maps should be static constants, no reason to create new maps each time we execute `renderDuration`